README.md

```diff
@@ -22,12 +22,20 @@ Unlike the Yocto BMAP tool, `bmap-writer` is C++ based does not require Python a
 1. Create a BMAP file for your disk image using `bmaptool`.
 2. Use `bmap-writer` to write the image to your target device, specifying the BMAP file for efficient block mapping.
 
-## Example Usage
+## Usage
 
 ```
-bmap-writer <image-file> <bmap-file> <target-device>
+bmap-writer [-b <max-buf-size>] [-v] <image-file> <bmap-file> <target-device>
 ```
 
+The `<max-buf-size>` parameter can be used to limit the write buffer size on systems with limited RAM.
```

Comment:
would be possible to calculate the available free memory instead ?

for example using sysinfo : https://man7.org/linux/man-pages/man2/sysinfo.2.html

I would like to keep the tool as simple as possible and not introduce much options (or not at all)

Reply:
Yes, that can be an option, but if during execution the free memory lowers significantly, memory exhaustion can still be a risk.

Since the tool is meant for the embedded world, having a non-mandatory option that can be used to fix an arbitrary maximum (thus allowing for example parallel writes on different flash devices) is IMO still useful.

The two approaches can also be used together: if the user does not specify manually a maximum value, the tool can auto-calculate it itself from free memory.

Reply:
I inserted the query of sysinfo in the updated MR - let me know what you think. Thjank you